gcovr/tests/test_gcovr.py

```diff
@@ -249,21 +258,25 @@ def pytest_generate_tests(metafunc):
                 continue
 
             marks = [
+                pytest.mark.skipif(
+                    name == "simple1-drive-subst" and not IS_WINDOWS,
+                    reason="drive substitution only available on windows",
+                ),
                 pytest.mark.xfail(
                     name == "exclude-throw-branches"
                     and format == "html"
-                    and is_windows,
+                    and IS_WINDOWS,
                     reason="branch coverage details seem to be platform-dependent",
                 ),
                 pytest.mark.xfail(
-                    name == "rounding" and is_windows,
+                    name == "rounding" and IS_WINDOWS,
                     reason="branch coverage seem to be platform-dependent",
                 ),
-                pytest.mark.xfail(
+                pytest.mark.skipif(
                     name == "html-source-encoding-cp1252" and IS_CLANG,
                     reason="clang doesnt understand -finput-charset=...",
                 ),
-                pytest.mark.xfail(
+                pytest.mark.skipif(
                     name == "gcc-abspath"
                     and (
                         not env["CC"].startswith("gcc-")
```

Comment:
These source-encoding test were xfail so that they would still run under Clang, in case future Clang versions add support for this command line option. But since the test suite pins certain Clang versions (and since this relates to an extremely niche feature), skipping that test entirely is fine.

On the other hand, the “gcc-abspath” test relates to the extremely useful `-fprofile-abs-path` option. It would be good to keep attempting this test, so that we are alerted if the option becomes available with Clang as well.